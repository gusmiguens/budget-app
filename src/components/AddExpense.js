const React = require('react')
const { connect } = require('react-redux')
const ExpenseForm = require('./ExpenseForm')
const { addExpense } = require('../actions/expenses')

class AddExpense extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense)
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    addExpense: (expense) => dispatch(addExpense(expense))
})

const connectedAddExpense = connect(undefined, mapDispatchToProps)(AddExpense)

module.exports = {
    connectedAddExpense,
    AddExpense
}