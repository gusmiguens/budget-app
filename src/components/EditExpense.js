const React = require('react')
const { connect } = require('react-redux')
const { editExpense, removeExpense } = require('../actions/expenses')
const ExpenseForm = require('./ExpenseForm')

class EditExpense extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/')
    }
    onRemove = () => {
        this.props.removeExpense({ id: this.props.expense.id });
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button 
                onClick={this.onRemove}
                >Remove Expense
                </button>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch, props) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (data) => dispatch(removeExpense(data))
})

const connectedEditExpense = connect(mapStateToProps, mapDispatchToProps)(EditExpense)

module.exports = {
    connectedEditExpense,
    EditExpense
}