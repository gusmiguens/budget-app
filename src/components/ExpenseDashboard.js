const React = require('react')
const { ExpenseList } = require('./ExpenseList')
const { ExpenseListFilters } = require('./ExpenseListFilters')

const ExpenseDashboard = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

module.exports = ExpenseDashboard