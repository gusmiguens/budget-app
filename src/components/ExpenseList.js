const React = require('react')
const { connect } = require('react-redux')
const ExpenseListItem = require('./ExpenseListItem')
const selectExpenses = require('../selectors/expenses')

const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                    props.expenses.map((expense) => {
                        return <ExpenseListItem key={expense.id} {...expense} />
                    })
                )
        }
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

const ConnectedExpenseList = connect(mapStateToProps)(ExpenseList)

module.exports = {
    ConnectedExpenseList,
    ExpenseList
}