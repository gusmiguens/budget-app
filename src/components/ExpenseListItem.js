const React = require('react')
const { Link } = require('react-router-dom')

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>Amount: {amount} - CreatedAt: {createdAt}</p>
    </div>
)

module.exports = ExpenseListItem