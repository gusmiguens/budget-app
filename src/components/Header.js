import React from 'react'
import { NavLink } from 'react-router-dom'

export default () => (
    <header>
        <h1>Budget</h1>
        <NavLink to='/' activeClassName='is-active' exact={true}>Dashboard</NavLink>
        <NavLink to='/create' activeClassName='is-active'>Create Page</NavLink>
    </header>
)