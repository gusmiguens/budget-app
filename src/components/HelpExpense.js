import React from 'react'

export default () => (
    <div>Help Expense</div>
)