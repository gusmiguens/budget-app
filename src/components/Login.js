import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'

export const Login = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Budget App</h1>
            <p>It's time to take control of your budget</p>
            <button className="button" onClick={startLogin}>Login with Google</button>
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(Login)