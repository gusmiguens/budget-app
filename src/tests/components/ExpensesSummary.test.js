import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

describe('', () => {
    it('Should correctly render Expenses Summary with one expense', () => {
        const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={2356} />)
        expect(wrapper).toMatchSnapshot()
    })
    it('Should correctly render Expenses Summary with multiple expenses', () => {
        const wrapper = shallow(<ExpensesSummary expenseCount={13} expensesTotal={45625123} />)
        expect(wrapper).toMatchSnapshot()
    })
})
