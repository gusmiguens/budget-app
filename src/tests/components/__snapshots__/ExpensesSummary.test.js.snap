// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render Expenses Summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    13
     
    expenses
     totalling 
    $456,251.23
  </h1>
</div>
`;

exports[`Should correctly render Expenses Summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $23.56
  </h1>
</div>
`;
